app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -277,6 +287,10 @@ public void onClick(View v) {
         if (!isUsingActionBar()) {
             searchbox.setText(lastQueryString);
         }
+
+        if (!this.containsHereFunction) {
+            hereFunctionHandler.refreshLocation();
+        }

```

Comment:
`refreshLocation` starts out false, and only gets set to true when `eval` is called. So in `onCreate`, the only time `refreshLocation` is true is when there's a screen rotation after `eval` has been called. And in that case, we don't want to reset the location, we just want to enable GPS/network use (which occurs in `onResume`).